avs_core/CMakeLists.txt

```diff
@@ -102,10 +102,16 @@ if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
     set(FSLIB "stdc++fs")
 endif()
 
-# Windows DLL dependencies
+# Per-OS link dependencies
 if (MSVC OR MINGW)
   target_link_libraries("AvsCore" "uuid" "winmm" "vfw32" "msacm32" "gdi32" "user32" "advapi32" "ole32" "imagehlp")
-else()
+endif()
+
+if (HAIKU)
+  target_link_libraries("AvsCore" "pthread" "${FSLIB}")
```

Comment:
Hello! Just a quick note: 
Haiku's libpthread is only a "fake library" for compatibility purposes, and instead, you'll want to link to libroot. In other words, you'll want to replace "pthread" with "root" on this line.

Reply:
I knew it was true for dlopen, that's why that one was omitted.  I don't remember if I'd seen that pthreads were also counted in that too, but thanks for confirming it.

Reply:
No problem!